flake.lock

```diff
@@ -0,0 +1,94 @@
+{
+  "nodes": {
+    "flake-compat": {
+      "flake": false,
+      "locked": {
+        "lastModified": 1650374568,
+        "narHash": "sha256-Z+s0J8/r907g149rllvwhb4pKi8Wam5ij0st8PwAh+E=",
+        "owner": "edolstra",
+        "repo": "flake-compat",
+        "rev": "b4a34015c698c7793d592d66adbab377907a2be8",
+        "type": "github"
+      },
+      "original": {
+        "owner": "edolstra",
+        "repo": "flake-compat",
+        "type": "github"
+      }
+    },
+    "nixpkgs": {
+      "locked": {
+        "lastModified": 1654230545,
+        "narHash": "sha256-8Vlwf0x8ow6pPOK2a04bT+pxIeRnM1+O0Xv9/CuDzRs=",
+        "owner": "nixos",
+        "repo": "nixpkgs",
+        "rev": "236cc2971ac72acd90f0ae3a797f9f83098b17ec",
+        "type": "github"
+      },
+      "original": {
+        "owner": "nixos",
+        "ref": "nixos-unstable",
+        "repo": "nixpkgs",
+        "type": "github"
+      }
+    },
```

Comment:
I want it to be explicit to the user which Nixpkgs they are using in their configuration. I.e., HM should never be responsible for pinning Nixpkgs on behalf of users. So if we are pinning Nixpkgs here wouldn't we also need to change

    pkgs ? builtins.getAttr system nixpkgs.outputs.legacyPackages

to simply

    pkgs

in the `homeManagerConfiguration` function?

Ideally, perhaps disallowing

    home-manager.url = "github:nix-community/home-manager";

without an accompanying

    home-manager.inputs.nixpkgs.follows = "nixpkgs";

would be most reassuring but I don't know if that is possible.

Reply:
> So if we are pinning Nixpkgs here wouldn't we also need to change

Agreed, I'll do that in a follow-up PR.

> Ideally, perhaps disallowing

That's not possible, and I think it would be overkill: we only need to ensure that the `nixpkgs` input is not passed down to the HM modules, but we may (and should) use the locked `nixpkgs` for flake packages.